pkg/buildpod/generator.go

```diff
@@ -24,12 +24,12 @@ func (g *Generator) Generate(build *v1alpha1.Build) (*v1.Pod, error) {
 
 func (g *Generator) getBuildSecrets(build *v1alpha1.Build) ([]corev1.Secret, error) {
 	var secrets []corev1.Secret
-	serviceAccount, err := g.K8sClient.CoreV1().ServiceAccounts(build.Namespace()).Get(build.ServiceAccount(), metav1.GetOptions{})
+	serviceAccount, err := g.K8sClient.CoreV1().ServiceAccounts(build.ObjectMeta.Namespace).Get(build.Spec.ServiceAccount, metav1.GetOptions{})
```

Comment:
I fell like the nicety of having a method that give you the namespace from the object directly instead of drilling through the objects to find it is lost here.
Why did you decide to remove the method?

Reply:
No particular reason, Build no-longer implements ImageRef so, I removed the Namespace method.

I think you are right that it is valuable on its own. I will put it back.

Reply:
Isnt Namespace just an embedded property on the Build type? Do we have to go through the objectMeta? or does it still have a method called Namespace() on it?

Reply:
You are right @ashwin-venkatesh. Just `.Namesspace` makes sense.

Reply:
Credit for this one should go to @djoyahoy as well